src/gui/symbology/qgssymbollayerwidget.cpp

```diff
@@ -323,6 +344,12 @@ void QgsSimpleLineSymbolLayerWidget::setSymbolLayer( QgsSymbolLayer *layer )
 
   whileBlocking( mRingFilterComboBox )->setCurrentIndex( mRingFilterComboBox->findData( mLayer->ringFilter() ) );
 
+  whileBlocking( mCheckAlignDash )->setChecked( mLayer->alignDashPattern() );
+  mCheckDashCorners->setEnabled( mLayer->alignDashPattern() );
+  whileBlocking( mCheckDashCorners )->setChecked( mLayer->tweakDashPatternOnCorners() );
+  if ( !mLayer->alignDashPattern() )
+    mCheckDashCorners->setChecked( false );
```

Comment:
Why not adding this condition in the whileBlocking( mCheckDashCorners )->setChecked( ... ) line above?

`whileBlocking( mCheckDashCorners )->setChecked( mLayer->tweakDashPatternOnCorners() && !mLayer->alignDashPattern() ) );`